### PR TITLE
feat: mock nested mappings

### PR DESCRIPTION
**Pull request by 0xJabberwock**
Closes BES-137

### CODE REVIEW

File: src/types.ts [hunk 1]
```diff
@@ -1,30 +1,29 @@
+export interface TypeNameNode {
+  nodeType?: 'ElementaryTypeName' | 'ArrayTypeName' | 'Mapping' | 'UserDefinedTypeName';
+  typeDescriptions?: {
```
Comment (0xJabberwock) [inline src/types.ts:3]: These properties are optional only for the purpose of test suite immutability.

File: src/types.ts [hunk 2]
```diff
@@ -1,30 +1,29 @@
...
-      };
-    };
-  };
+  typeName?: TypeNameNode;
+  name?: string;
+  constant?: boolean;
+  mutability?: string;
+  visibility?: string;
+  stateVariable?: boolean;
+  storageLocation?: string;
```
Comment (0xJabberwock) [inline src/types.ts:26]: These properties were and remain being optional only for test suite convenience.

File: src/types.ts [hunk 3]
```diff
@@ -97,13 +97,12 @@ export interface BasicStateVariableOptions {
 export interface MappingStateVariableOptions {
   setFunction: {
     functionName: string;
-    keyType: string;
+    keyTypes: string[];
     valueType: string;
-    mappingName: string;
```
Comment (0xJabberwock) [inline src/types.ts:103]: `mappingName` property appeared to be redundant with `functionName`, which was the one preserved due to cohesion with the other `VariableOptions` types.

File: test/unit/set-variables.spec.ts [hunk 4]
```diff
@@ -30,15 +30,14 @@ describe('getStateVariables', () => {
...
-  it('should return the correct variable data if the variable is a base struct variable', async () => {
+  it('should return the correct data if the variable is a base struct', async () => {
     contractNode.nodes = [
       {
         constant: false,
         mutability: 'mutable',
         visibility: 'public',
         name: 'myStructVariable',
         nodeType: 'VariableDeclaration',
-        storageLocation: 'default',
```
Comment (0xJabberwock) [inline test/unit/set-variables.spec.ts:41]: `storageLocation` property is never accessed across `getStateVariables()`.

**Review by gas1cent — APPROVED** [review]
Love it!

**Pull request merged**